flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/expressions/resolver/LookupCallResolver.java

```diff
@@ -63,6 +64,17 @@ public Expression visit(UnresolvedCallExpression unresolvedCall) {
 			.collect(Collectors.toList());
 	}
 
+	@Override
+	public Expression visitNonApiExpression(Expression other) {
+		// LookupCallResolver might be called outside of ExpressionResolver, thus we need to additionally
```

Comment:
I think it's not that simple. We call the `LookupCallResolver` externally primarily to separate functions into scalar and aggregating functions so that we can split expressions into ones that go into groupBy clause and those that go into select clause. Therefore before that step we might not have an input we can use to fully resolve expressions using `ExpressionResolver`.